src/main/java/com/icthh/xm/ms/balance/service/BalanceService.java

```diff
@@ -218,12 +234,25 @@ public BalanceChangeEventDto charging(ChargingBalanceRequest chargingRequest) {
     @Transactional
     @LogicExtensionPoint(value = "Charging", resolver = BalanceTypeKeyResolver.class)
     public BalanceChangeEventDto charging(Balance balance, ChargingBalanceRequest chargingRequest) {
-        assertBalanceAmout(balance, chargingRequest.getAmount());
+        assertBalanceAmount(balance, chargingRequest.getAmount());
+
+        String operationUuid = chargingRequest.getUuid();
+        if (StringUtils.isNotBlank(operationUuid)) {
```

Comment:
if (StringUtils.isNotBlank(operationUuid)) { //(with this check important!)
            List<BalanceChangeEvent> existBalanceChangeEvents =
                balanceChangeEventRepository.findBalanceChangeEventsByOperationId(operationUuid);
                
                extract to getBalanceChangeEvent